src/Form/AuthenticationForm.php

```diff
@@ -372,26 +392,187 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
+      elseif ($exception->getCode() === 403) {
+        $suggestion = $this->t('@fail_text The given organization name (%organization) is probably incorrect.', [
+          '@fail_text' => $fail_text,
+          '%organization' => $key_type->getOrganization($key),
+        ]);
+      }
+      // Failed request.
+      elseif ($exception->getCode() === 0) {
+        if ($exception->getPrevious() instanceof NetworkException && $exception->getPrevious()->getPrevious() instanceof ConnectException) {
+          /** @var \GuzzleHttp\Exception\ConnectException $curl_exception */
+          $curl_exception = $exception->getPrevious()->getPrevious();
+          // Resolving timed out.
+          if ($curl_exception->getHandlerContext()['errno'] === 28) {
+            $suggestion = $this->t('@fail_text Maybe the connection timeout threshold (%connect_timeout) or the request timeout (%timeout) is too low or something is wrong with the connection.', [
+              '@fail_text' => $fail_text,
+              '%connect_timeout' => $this->state->get('apigee_edge.client')['http_client_connect_timeout'],
+              '%timeout' => $this->state->get('apigee_edge.client')['http_client_timeout'],
+            ]);
+          }
+          // The remote host was not resolved (endpoint).
+          elseif ($curl_exception->getHandlerContext()['errno'] === 6) {
+            $suggestion = $this->t('@fail_text The given endpoint (%endpoint) is probably incorrect or something is wrong with the connection.', [
+              '@fail_text' => $fail_text,
+              '%endpoint' => $key_type->getEndpoint($key),
+            ]);
+          }
+        }
+      }
+    }
+
+    return $suggestion;
+  }
+
+  /**
+   * Creates debug text if there was an error during form validation.
+   *
+   * @param \Exception $exception
+   *   The thrown exception during form validation.
+   * @param \Drupal\key\KeyInterface $key
+   *   The used key during form validation.
+   * @param \Drupal\key\KeyInterface|null $key_token
+   *   The user token key during form validation.
+   *
+   * @return string
+   *   The debug text to be displayed.
+   */
+  protected function createDebugText(\Exception $exception, KeyInterface $key, ?KeyInterface $key_token): string {
+    /** @var \Drupal\apigee_edge\Plugin\KeyType\BasicAuthKeyType $key_type */
+    $key_type = $key->getKeyType();
+
+    $credentials = [
+      'endpoint' => $key_type->getEndpoint($key) ?? ClientInterface::DEFAULT_ENDPOINT,
+      'organization' => $key_type->getOrganization($key),
+      'username' => $key_type->getUsername($key),
+    ];
+
+    $keys = [
+      'key_type' => get_class($key_type),
+      'key_provider' => get_class($key->getKeyProvider()),
+    ];
+
+    if ($key_type instanceof OauthKeyType) {
+      /** @var \Drupal\apigee_edge\Plugin\KeyType\OauthKeyType $key_type */
+      $credentials['authorization_server'] = $key_type->getAuthorizationServer($key) ?? Oauth::DEFAULT_AUTHORIZATION_SERVER;
```

Comment:
Why `getAuthorizationServer()` does not return `Oauth::DEFAULT_AUTHORIZATION_SERVER` if it has not been changed to something else? The same question for `getClientId()` and `getClientSecret()`. Would it be complicated to change the implementation? (IIRC we talked about something like this before)